### PR TITLE
Proposal for way to ignore rules via comments #49

### PR DESCRIPTION
**Pull request by thepanu**
Proposal to how offenses could be ignored per definition.

### CODE REVIEW

**Comment by DamirSvrtan** [comment]
[Ignore rules](https://trello.com/c/fNRHDs6k/2-ignore-rules)

**Comment by volkanunsal** [comment]
Any progress in merging this?

**Comment by DamirSvrtan** [comment]
I am sorry, this totally went under my radar. I'll leave some comments, @thepanu I thank you very much for your work on this!

File: lib/fasterer/analyzer.rb [hunk 1]
```diff
@@ -26,15 +27,20 @@ def errors
...
     private
 
     def traverse_sexp_tree(sexp_tree)
       return unless sexp_tree.is_a?(Sexp)
 
       token = sexp_tree.first
-
       scan_by_token(token, sexp_tree)
 
+      comment.new_comment(sexp_tree.comments) if token == :defn
```
Comment (DamirSvrtan) [inline lib/fasterer/analyzer.rb]: This only ignores comments on a method definition? What about other cases?

File: spec/support/analyzer/97_ignore_by_comment.rb [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+end
+
+# :rescue_vs_respond_to
+def ignored_rescue_vs_respondd
+  begin
+    'abakus'.to_a
+  rescue NoMethodError
+
+  end
+end
```
Comment (DamirSvrtan) [inline spec/support/analyzer/97_ignore_by_comment.rb]: what would happen if there was a third method that rescued the no method error as well but was not ignored?

File: spec/support/analyzer/97_ignore_by_comment.rb [hunk 3]
```diff
@@ -0,0 +1,16 @@
+def not_ignored_rescue_vs_respond
+  begin
+    'abakus'.to_a
+  rescue NoMethodError
+
+  end
+end
+
+# :rescue_vs_respond_to
```
Comment (DamirSvrtan) [inline spec/support/analyzer/97_ignore_by_comment.rb]: Would it make sense to follow Rubocop's naming convention:

```ruby
# rubocop:disable RuleByName
```

e.g.

```ruby
# fasterer:disable rescue_vs_respond_to
```

File: spec/support/analyzer/97_ignore_by_comment.rb (hunk 2 above)
Comment (DamirSvrtan) [reply spec/support/analyzer/97_ignore_by_comment.rb]: I guess the third one would be accounted for as there is the reset_comment method.

**Comment by DamirSvrtan** [comment]
@thepanu any interest in continuing with this feature work or should we close the branch for now?

**Comment by thepanu** [comment]
I think it is safe to close for now, haven't touched any code in a bit.

**Comment by DamirSvrtan** [comment]
Closing as we're not progressing on this. For anyone that's willing to take this work on, this PR would be a good starting point.

**Pull request closed without merging**